java/bogdana_maksiuta/src/com/zoolatech/lecture2/tasks/_1/Calculator.java

```diff
@@ -0,0 +1,70 @@
+package com.zoolatech.lecture2.tasks._1;
+
+import java.util.Scanner;
+
+public class Calculator {
+    Scanner scanner = new Scanner(System.in);
+
+    public void processOfCalculation() {
+        double firstNumber = inputNumber();
+        char symbol = inputSymbol();
+        double secondNumber = inputNumber();
+
+        switch (symbol) {
+            case '+' -> addition(firstNumber, secondNumber);
+            case '-' -> subtraction(firstNumber, secondNumber);
+            case '*' -> multiplication(firstNumber, secondNumber);
+            case '/' -> division(firstNumber, secondNumber);
+            default -> System.out.println("Invalid symbol");
+        }
+    }
+
+    public double inputNumber() {
+        System.out.println("Please enter a number : ");
+        double number;
+        if (scanner.hasNextDouble()) {
+            number = scanner.nextDouble();
+        } else {
+            System.out.println("Invalid value. Try another number : ");
+            scanner.next();
+            number = inputNumber();
```

Comment:
Interesting idea with the recursion. However, prefer to use a loop over the recursion since the later one has a limited number of operations (it's hard to receive it practically, but still) and uses more memory